mobly/controllers/android_device.py

```diff
@@ -995,16 +1000,17 @@ def take_bug_report(self,
     self.log.debug('Bugreport taken at %s.', full_out_path)
     return full_out_path
 
-  def take_screenshot(self, destination):
+  def take_screenshot(self, destination, filename=None):
     """Takes a screenshot of the device.
 
```

Comment:
explain in docstring the detailed behavior of the `filename`
what happens if no filename is given etc.